App crash on startup

**Describe the bug**
Trying to open the application cause an unexpected exit.

**Environment information:**

- OS: Linux
- Platform: Linux 5.2.0-2-amd64 x86_64
- Go: go version go1.12.9 linux/amd64

**Steps to Reproduce**
Steps to reproduce the behavior:

1. `make run` (without a skycoin node running)
2. See error.

**Actual behavior**
Crash

**Expected behavior**
Show the application.

**Additional context**
```
[2019-10-29T12:55:07-04:00] ERROR [Skycoin Account]: Couldn't POST /api/v1/transactions?verbose=1 addrs="rspabT3fQVPa1bv3Z2fQFyw84gbEibjX3w" error="Get http://127.0.0.1:6420/api/v1/csrf: dial tcp 127.0.0.1:6420: connect: connection refused"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xc1b48b]

goroutine 1 [running, locked to thread]:
github.com/fibercrypto/FiberCryptoWallet/src/models/history.(*HistoryManager).getTransactionsOfAddresses(0xc000377200, 0xc00043c800, 0xb, 0x10, 0x9, 0xc00043c800, 0x8)
        github.com/fibercrypto/FiberCryptoWallet/src/models/history/historyManager.go:93 +0x3cb
github.com/fibercrypto/FiberCryptoWallet/src/models/history.(*HistoryManager).loadHistory(0xc000377200, 0x0, 0xc00040a700, 0x2edb390)
        github.com/fibercrypto/FiberCryptoWallet/src/models/history/historyManager.go:344 +0x1d1
github.com/fibercrypto/FiberCryptoWallet/src/models/history.callbackHistoryManager175bf8_LoadHistory.func1(0xc000254918, 0x2f03c50)
        github.com/fibercrypto/FiberCryptoWallet/src/models/history/moc.go:157 +0x61
github.com/fibercrypto/FiberCryptoWallet/src/models/history.callbackHistoryManager175bf8_LoadHistory(0x2f03c50, 0xc0000cdc88)
        github.com/fibercrypto/FiberCryptoWallet/src/models/history/moc.go:161 +0x5f
github.com/fibercrypto/FiberCryptoWallet/src/models/history._cgoexpwrap_da504d2b9cce_callbackHistoryManager175bf8_LoadHistory(0x2f03c50, 0x0)
        _cgo_gotypes.go:3083 +0x64
github.com/therecipe/qt/qml._Cfunc_QQmlApplicationEngine_Load(0x22d2890, 0x23751e0)
        _cgo_gotypes.go:921 +0x45
github.com/therecipe/qt/qml.(*QQmlApplicationEngine).Load.func1(0xc0002540d8, 0x104c560, 0xc0002540f0)
        github.com/therecipe/qt/qml/qml-minimal.go:792 +0xb4
github.com/therecipe/qt/qml.(*QQmlApplicationEngine).Load(0xc0002540d8, 0x104c560, 0xc0002540f0)
        github.com/therecipe/qt/qml/qml-minimal.go:792 +0x59
main.main()
        github.com/fibercrypto/FiberCryptoWallet/main.go:41 +0x1fc
make: *** [Makefile:14: run] Error 2
```

--- a/src/models/history/historyManager.go
+++ b/src/models/history/historyManager.go
@@ -90,6 +90,10 @@ func (hm *HistoryManager) getTransactionsOfAddresses(filterAddresses []string) [
 			_, ok := find[addressIterator.Value().String()]
 			if ok {
 				txnsIterator := addressIterator.Value().GetCryptoAccount().ListTransactions()
+				if txnsIterator == nil {
+					logHistoryManager.Warn("Couldn't get transaction iterator")
+					continue
+				}
 				for txnsIterator.Next() {
 					_, ok2 := txnFind[txnsIterator.Value().GetId()]
 					if !ok2 {

--- a/src/models/history/historyManager.go
+++ b/src/models/history/historyManager.go
@@ -85,7 +85,11 @@ func (hm *HistoryManager) getTransactionsOfAddresses(filterAddresses []string) [
 		return make([]*transactions.TransactionDetails,0)
 	}
 	for wltIterator.Next() {
-		addressIterator, _ := wltIterator.Value().GetLoadedAddresses()
+		addressIterator, err := wltIterator.Value().GetLoadedAddresses()
+		if err != nil {
+			logHistoryManager.Warn("Couldn't get address iterator")
+			continue
+		}
 		for addressIterator.Next() {
 			_, ok := find[addressIterator.Value().String()]
 			if ok {

--- a/src/models/walletsManager.go
+++ b/src/models/walletsManager.go
@@ -2,6 +2,7 @@ package models
 
 import (
 	"errors"
+	"fmt"
 	"github.com/fibercrypto/FiberCryptoWallet/src/coin/skycoin/params"
 	"sync"
 
@@ -134,30 +135,28 @@ func (walletM *WalletManager) updateAddresses(wltId string) {
 		qAddress.SetWalletId(wlt.GetId())
 		skyFl, err := addr.GetCryptoAccount().GetBalance(params.SkycoinTicker)
 		if err != nil {
-			qAddress.SetAddressSky("0")
-			qAddress.SetAddressCoinHours("0")
+			qAddress.SetAddressSky("N/A")
+			qAddress.SetAddressCoinHours("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.SkycoinTicker + " Balance")
 			continue
 		}
 		accuracy, err := util.AltcoinQuotient(params.SkycoinTicker)
 		if err != nil {
-			qAddress.SetAddressSky("0")
-			qAddress.SetAddressCoinHours("0")
+			qAddress.SetAddressSky("N/A")
+			qAddress.SetAddressCoinHours("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.SkycoinTicker + " quotient")
 			continue
 		}
 		qAddress.SetAddressSky(util.FormatCoins(skyFl, accuracy))
 		coinH, err := addr.GetCryptoAccount().GetBalance(params.CoinHoursTicker)
 		if err != nil {
-			qAddress.SetAddressSky("0")
-			qAddress.SetAddressCoinHours("0")
+			qAddress.SetAddressCoinHours("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.CoinHoursTicker + " Balance")
 			continue
 		}
 		accuracy, err = util.AltcoinQuotient(params.CoinHoursTicker)
 		if err != nil {
-			qAddress.SetAddressSky("0")
-			qAddress.SetAddressCoinHours("0")
+			qAddress.SetAddressCoinHours("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.CoinHoursTicker + " quotient")
 			continue
 		}
@@ -203,13 +202,13 @@ func (walletM *WalletManager) updateOutputs(wltId, address string) {
 		qout.SetOutputID(outsIter.Value().GetId())
 		skyV, err := outsIter.Value().GetCoins(sky.Sky)
 		if err != nil {
-			qout.SetAddressSky("0")
+			qout.SetAddressSky("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.Sky + " coins")
 			continue
 		}
 		quotient, err := util.AltcoinQuotient(sky.Sky)
 		if err != nil {
-			qout.SetAddressSky("0")
+			qout.SetAddressSky("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.Sky + " quotient")
 			continue
 		}
@@ -217,13 +216,13 @@ func (walletM *WalletManager) updateOutputs(wltId, address string) {
 		qout.SetAddressSky(sSky)
 		ch, err := outsIter.Value().GetCoins(sky.CoinHour)
 		if err != nil {
-			qout.SetAddressCoinHours("0")
+			qout.SetAddressCoinHours("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.CoinHour + " coins")
 			continue
 		}
 		quotient, err = util.AltcoinQuotient(sky.CoinHour)
 		if err != nil {
-			qout.SetAddressCoinHours("0")
+			qout.SetAddressCoinHours("N/A")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.Sky + " quotient")
 			continue
 		}
@@ -653,30 +652,30 @@ func fromWalletToQWallet(wlt core.Wallet, isEncrypted bool) *QWallet {
 
 	bl, err := wlt.GetCryptoAccount().GetBalance(sky.Sky)
 	if err != nil {
-		qWallet.SetSky(0)
-		qWallet.SetCoinHours(0)
+		qWallet.SetSky("N/A")
+		qWallet.SetCoinHours("N/A")
 		logWalletManager.WithError(err).Error("Couldn't get Skycoin balance")
 		return qWallet
 	}
 
 	accuracy, err := util.AltcoinQuotient(params.SkycoinTicker)
 	if err != nil {
-		qWallet.SetSky(0)
-		qWallet.SetCoinHours(0)
+		qWallet.SetSky("N/A")
+		qWallet.SetCoinHours("N/A")
 		logWalletManager.WithError(err).Error("Couldn't get Skycoin Altcoin quotient")
 		return qWallet
 	}
 
 	floatBl := float64(bl) / float64(accuracy)
-	qWallet.SetSky(floatBl)
+	qWallet.SetSky(fmt.Sprintf("%f", floatBl))
 
 	bl, err = wlt.GetCryptoAccount().GetBalance(sky.CoinHoursTicker)
 	if err != nil {
-		qWallet.SetCoinHours(0)
+		qWallet.SetCoinHours("N/A")
 		logWalletManager.WithError(err).Error("Couldn't get Coin Hours balance")
 		return qWallet
 	}
-	qWallet.SetCoinHours(bl)
+	qWallet.SetCoinHours(fmt.Sprintf("%f", bl))
 
 	return qWallet
 }

--- a/src/models/walletsManager.go
+++ b/src/models/walletsManager.go
@@ -134,22 +134,30 @@ func (walletM *WalletManager) updateAddresses(wltId string) {
 		qAddress.SetWalletId(wlt.GetId())
 		skyFl, err := addr.GetCryptoAccount().GetBalance(params.SkycoinTicker)
 		if err != nil {
+			qAddress.SetAddressSky("0")
+			qAddress.SetAddressCoinHours("0")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.SkycoinTicker + " Balance")
 			continue
 		}
 		accuracy, err := util.AltcoinQuotient(params.SkycoinTicker)
 		if err != nil {
+			qAddress.SetAddressSky("0")
+			qAddress.SetAddressCoinHours("0")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.SkycoinTicker + " quotient")
 			continue
 		}
 		qAddress.SetAddressSky(util.FormatCoins(skyFl, accuracy))
 		coinH, err := addr.GetCryptoAccount().GetBalance(params.CoinHoursTicker)
 		if err != nil {
+			qAddress.SetAddressSky("0")
+			qAddress.SetAddressCoinHours("0")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.CoinHoursTicker + " Balance")
 			continue
 		}
 		accuracy, err = util.AltcoinQuotient(params.CoinHoursTicker)
 		if err != nil {
+			qAddress.SetAddressSky("0")
+			qAddress.SetAddressCoinHours("0")
 			logWalletManager.WithError(err).Warn("Couldn't load " + params.CoinHoursTicker + " quotient")
 			continue
 		}
@@ -195,11 +203,13 @@ func (walletM *WalletManager) updateOutputs(wltId, address string) {
 		qout.SetOutputID(outsIter.Value().GetId())
 		skyV, err := outsIter.Value().GetCoins(sky.Sky)
 		if err != nil {
+			qout.SetAddressSky("0")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.Sky + " coins")
 			continue
 		}
 		quotient, err := util.AltcoinQuotient(sky.Sky)
 		if err != nil {
+			qout.SetAddressSky("0")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.Sky + " quotient")
 			continue
 		}
@@ -207,11 +217,13 @@ func (walletM *WalletManager) updateOutputs(wltId, address string) {
 		qout.SetAddressSky(sSky)
 		ch, err := outsIter.Value().GetCoins(sky.CoinHour)
 		if err != nil {
+			qout.SetAddressCoinHours("0")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.CoinHour + " coins")
 			continue
 		}
 		quotient, err = util.AltcoinQuotient(sky.CoinHour)
 		if err != nil {
+			qout.SetAddressCoinHours("0")
 			logWalletManager.WithError(err).Warn("Couldn't get " + sky.Sky + " quotient")
 			continue
 		}
@@ -239,13 +251,8 @@ func (walletM *WalletManager) updateWallets() {
 			logWalletManager.WithError(err).Warn("Couldn't get wallet by id")
 			continue
 		}
-		if encrypted {
-			qw := fromWalletToQWallet(it.Value(), true)
-			qWallets = append(qWallets, qw)
-		} else {
-			qw := fromWalletToQWallet(it.Value(), false)
-			qWallets = append(qWallets, qw)
-		}
+		qw := fromWalletToQWallet(it.Value(), encrypted)
+		qWallets = append(qWallets, qw)
 
 	}
 	walletM.wallets = qWallets
@@ -646,18 +653,16 @@ func fromWalletToQWallet(wlt core.Wallet, isEncrypted bool) *QWallet {
 
 	bl, err := wlt.GetCryptoAccount().GetBalance(sky.Sky)
 	if err != nil {
+		qWallet.SetSky(0)
+		qWallet.SetCoinHours(0)
 		logWalletManager.WithError(err).Error("Couldn't get Skycoin balance")
 		return qWallet
 	}
 
 	accuracy, err := util.AltcoinQuotient(params.SkycoinTicker)
 	if err != nil {
-		logWalletManager.WithError(err).Error("Couldn't get Skycoin Altcoin quotient")
-		return qWallet
-	}
-
-	accuracy, err = util.AltcoinQuotient(params.SkycoinTicker)
-	if err != nil {
+		qWallet.SetSky(0)
+		qWallet.SetCoinHours(0)
 		logWalletManager.WithError(err).Error("Couldn't get Skycoin Altcoin quotient")
 		return qWallet
 	}
@@ -667,12 +672,8 @@ func fromWalletToQWallet(wlt core.Wallet, isEncrypted bool) *QWallet {
 
 	bl, err = wlt.GetCryptoAccount().GetBalance(sky.CoinHoursTicker)
 	if err != nil {
+		qWallet.SetCoinHours(0)
 		logWalletManager.WithError(err).Error("Couldn't get Coin Hours balance")
-		return qWallet
-	}
-	accuracy, err = util.AltcoinQuotient(sky.SkycoinTicker)
-	if err != nil {
-		logWalletManager.WithError(err).Error("Couldn't get Coin Hours Altcoin quotient")
 		return qWallet
 	}
 	qWallet.SetCoinHours(bl)

--- a/src/models/walletsManager.go
+++ b/src/models/walletsManager.go
@@ -667,7 +667,7 @@ func fromWalletToQWallet(wlt core.Wallet, isEncrypted bool) *QWallet {
 	}
 
 	floatBl := float64(bl) / float64(accuracy)
-	qWallet.SetSky(fmt.Sprintf("%f", floatBl))
+	qWallet.SetSky(fmt.Sprint(floatBl))
 
 	bl, err = wlt.GetCryptoAccount().GetBalance(sky.CoinHoursTicker)
 	if err != nil {
@@ -675,7 +675,7 @@ func fromWalletToQWallet(wlt core.Wallet, isEncrypted bool) *QWallet {
 		logWalletManager.WithError(err).Error("Couldn't get Coin Hours balance")
 		return qWallet
 	}
-	qWallet.SetCoinHours(fmt.Sprintf("%f", bl))
+	qWallet.SetCoinHours(fmt.Sprint(bl))
 
 	return qWallet
 }

--- a/src/models/walletsManager.go
+++ b/src/models/walletsManager.go
@@ -1,7 +1,6 @@
 package models
 
 import (
-	"errors"
 	"fmt"
 	"github.com/fibercrypto/FiberCryptoWallet/src/coin/skycoin/params"
 	"sync"

--- a/src/models/walletsModel.go
+++ b/src/models/walletsModel.go
@@ -25,7 +25,7 @@ type WalletModel struct {
 	_ []*QWallet               `property:"wallets"`
 
 	_ func(*QWallet)                                                                    `slot:"addWallet"`
-	_ func(row int, name string, encryptionEnabled bool, sky float64, coinHours uint64) `slot:"editWallet"`
+	_ func(row int, name string, encryptionEnabled bool, sky string, coinHours string)  `slot:"editWallet"`
 	_ func(row int)                                                                     `slot:"removeWallet"`
 	_ func([]*QWallet)                                                                  `slot:"loadModel"`
 	_ int                                                                               `property:"count"`
@@ -36,8 +36,8 @@ type QWallet struct {
 
 	_ string  `property:"name"`
 	_ int     `property:"encryptionEnabled"`
-	_ float64 `property:"sky"`
-	_ uint64  `property:"coinHours"`
+	_ string  `property:"sky"`
+	_ string  `property:"coinHours"`
 	_ string  `property:"fileName"`
 }
 
@@ -129,7 +129,7 @@ func (walletModel *WalletModel) addWallet(w *QWallet) {
 
 }
 
-func (walletModel *WalletModel) editWallet(row int, name string, encrypted bool, sky float64, coinHours uint64) {
+func (walletModel *WalletModel) editWallet(row int, name string, encrypted bool, sky string, coinHours string) {
 	logWalletsModel.Info("Edit Wallet")
 	w := walletModel.Wallets()[row]
 	w.SetName(name)

--- a/src/ui/PageWallets.qml
+++ b/src/ui/PageWallets.qml
@@ -32,7 +32,7 @@ Page {
                 Layout.fillWidth: true
             }
             Label {
-                text: qsTr("Sky")
+                text: qsTr("SKY")
                 font.pointSize: 9
                 horizontalAlignment: Text.AlignRight
                 Layout.preferredWidth: internalLabelsWidth

